api/types/types_test.go

```diff
@@ -0,0 +1,43 @@
+/*
+Copyright 2021 Gravitational, Inc.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package types
+
+import (
+	"testing"
+
+	"github.com/stretchr/testify/require"
+)
+
+// Check that default constructors never return nil and pass validation checks.
+func TestDefaultConstructors(t *testing.T) {
+	require.NotNil(t, DefaultAuthPreference())
```

Comment:
The code could be perhaps shortened using something like
```go
defaultResources := []Resource{DefaultAuthPreference(), DefaultClusterAuditConfig(), DefaultClusterConfig(), DefaultClusterNetworkingConfig(), DefaultSessionRecordingConfig(), DefaultStaticTokens()}
for _, res := range defaultResources {
	require.NotNil(t, res)
	require.NoError(t, res.CheckAndSetDefaults())
}
```
but otherwise I wouldn't bother too much with this.